core/src/test/java/dk/alexandra/fresco/framework/network/async/TestAsyncNetwork.java

```diff
@@ -0,0 +1,316 @@
+package dk.alexandra.fresco.framework.network.async;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import dk.alexandra.fresco.framework.Party;
+import dk.alexandra.fresco.framework.configuration.NetworkConfiguration;
+import dk.alexandra.fresco.framework.configuration.NetworkConfigurationImpl;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.nio.channels.ServerSocketChannel;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.FutureTask;
+import javax.net.ServerSocketFactory;
+import org.junit.Test;
+
+public class TestAsyncNetwork {
```

Comment:
General comments to the tests:

- I think some tests are missing here. It would be nice if the network is quite well tested. For example all tests here seem to be for 2 parties, and I do not see any tests of sending/receiving more than a single message. Maybe think about some of the ways the KryoNet implementation failed and add tests for that also. 
- It looks like this copies at lot from the KryoNet tests. It would be great to generalize the non-implementation specific tests into a set of tests which we can run on any new `Network` implementation.